math/stanc3/Portfile

```diff
@@ -4,19 +4,18 @@ PortSystem              1.0
 PortGroup               github 1.0
 PortGroup               ocaml 1.1
 
-# https://github.com/stan-dev/stanc3/issues/1371
-github.setup            stan-dev stanc3 02a3a856b45dcd9afcc72f8539d5e45430cec70b
-version                 2023.12.08
+github.setup            stan-dev stanc3 2.34.0 v
+epoch                   1
 revision                0
 categories              math lang ocaml
 maintainers             {@barracuda156 gmail.com:vital.had} {@catap korins.ky:kirill} openmaintainer
 license                 BSD
 description             New compiler for Stan, written in OCaml
 long_description        {*}${description}
 homepage                https://mc-stan.org/stanc3/stanc
-checksums               rmd160  357efbe291081693f9e9384c7798f7fd7d4f29ff \
-                        sha256  18cb2ce55d736f4678ad52e2fbb36a4fb16b78b05e46b0b7ba3c456da20dfeaa \
-                        size    3630544
+checksums               rmd160  f38cffbbc29bc59dddd08e0bc5fce85566fb459f \
+                        sha256  ef7624d01fdced7834c1cc6afaf55cb61bcc468499267e05fee8e256985b36eb \
+                        size    3583890
 github.tarball_from     archive
 
 set py_ver              3.11
```

Comment:
Please se Python 3.12 if possible

Reply:
This is not a Python port, this is OCaml port. Python is handling some scripts, and 3.11 can do that perfectly fine.
There is neither a necessity nor anything to gain by a switch.
(Python 3.12 is still broken for PPC in the master, and adding an extra chunk of code to use Python versions conditionally is unjustified, given its role here.)

Reply:
I haven’t heard anyone complaining about Python 3.11 suddenly not working on x86_64 or arm64 either, I believe :) 

I will add a conditional code now, but for the record, default does not imply it should be forced against maintainer’s preferences. As far as I can see, it is not used universally across every port either, and that is the case without any concern for PowerPC or other exotic systems.

Reply:
@reneeotten I suggest we stay here with the Python version which works on every system and do not break anything, even non-mainstream configurations. As soon as I come up with a solution to fix 3.12 in a way which gets accepts (likely following Ken’s suggestion), I will pro-actively move everything to 3.12. As of now, it does not make anyone better off if the port is not updated – sure enough, we can close the PR as long as this is not agreed upon, but then we still have 3.11 and outdated port.

This is a port with no dependencies at the moment, and if I make a port for CmdStan, it will still be a quite isolated thing. Nobody is adversely affected by having it at 3.11. Users who want to have Stan compiler, can bear with 3.11 for a while, as they do anyway for multiple other ports of different maintainers.

Reply:
this is not reasonable

Reply:
you want to hold back all your ports to python 3.11 because of 10.6-ppc, take that to the dev mailing list.

You’ll get zero support there, I suspect.

Reply:
@kencu Thank you for merging, Ken.

Re Python: first of all, I intend to find a solution that will be satisfying to everyone. Of course, keeping to hold back is _not_ a sustainable option; so no, this is not what I plan to do. However, I have very few ports that use Python, and of those some can be moved to 3.12 without any concern for PowerPC (either because they do not work properly there anyway or because they depend on other ports which use 3.12 anyway, so it makes no difference). So for the context, we talk about literally a dozen of ports, perhaps, all of which are rather obscure ones in math category. There is no urgency with those, I believe. However, as I said, I do not consider staying on 3.11 Python as a long-term option. Allow me some time and mood to think of reasonable solutions here.